src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -323,6 +323,17 @@ class Patcher : public RelocActions {
     offset >>= 2;
     Instruction_aarch64::spatch(insn_addr, 23, 5, offset);
     Instruction_aarch64::patch(insn_addr, 30, 29, offset_lo);
+
+    // Maybe we have a third instruction: adrp + movk + ldr with offset (e.g. ldr_patchable)
+    uint32_t insn3 = insn_at(insn_addr, 2);
```

Comment:
As discussed below, it's better not to do this. `adrp()` was never intended to be used in this way, and it's not so efficient to use adrp + movk + ldr with offset rather than movz; movk; movk; ldr (no offset).